Add some reasons for why binary mode might be used (#1035)
cloudevents/spec: cloudevents/spec.md

a stand-alone event format and stored in the message body.

A "binary-mode message" is one where the event data is stored in the message
body, and event attributes are stored as part of message metadata.

Often, binary mode is used when the producer of the CloudEvent wishes to add

> the CloudEvents metadata to an existing event without impacting the message's

Reviewer: ```suggestion
the CloudEvent's metadata to an existing event without impacting the message
```

Author: Done

> body. This means that in most cases a binary CloudEvents enabled message would

Reviewer: ```suggestion
body. In most cases a CloudEvent encoded as a binary-mode message will
```
I guess it's not just typos - I can't help wordsmithing sometimes - take it or not.

Author: done!

not break an existing receiver's processing of the event because the message's
metadata typically allows for extension attributes. In other words, a binary
formatted CloudEvent would work for both a CloudEvents enabled receiver as well

> as one that is unaware of CloudEvents.

Reviewer 2: Thanks @duglin - this is exactly the use case I had in mind!
